django/contrib/admin/checks.py

```diff
@@ -88,17 +104,16 @@ def check_dependencies(**kwargs):
                 "the admin application.",
                 id='admin.E404',
             ))
-    if ('django.contrib.auth.middleware.AuthenticationMiddleware'
-            not in settings.MIDDLEWARE):
+
+    if not _contains_subclass('django.contrib.auth.middleware.AuthenticationMiddleware', settings.MIDDLEWARE):
         errors.append(checks.Error(
-            "'django.contrib.auth.middleware.AuthenticationMiddleware' must "
+            "'django.contrib.auth.middleware.AuthenticationMiddleware' (or a subclass) must "
             "be in MIDDLEWARE in order to use the admin application.",
             id='admin.E408',
         ))
-    if ('django.contrib.messages.middleware.MessageMiddleware'
-            not in settings.MIDDLEWARE):
+    if not _contains_subclass('django.contrib.messages.middleware.MessageMiddleware', settings.MIDDLEWARE):
         errors.append(checks.Error(
-            "'django.contrib.messages.middleware.MessageMiddleware' must "
+            "'django.contrib.messages.middleware.MessageMiddleware' (or a subclass) must "
```

Comment:
Please wrap at 79 chars.

Reply:
Please update error message for `admin.E409` in `docs/ref/checks.txt`.

Reply:
Hey @felixxm. In the end we went for not having the "or subclassed" so these changes will disappear. (`checks.txt` did previously have this adjustment, but it's gone now. These are just leftovers to be removed.)

Reply:
@carltongibson Ahh sorry :man_facepalming: I should check discussion.